src/wp-includes/option.php

```diff
@@ -1133,6 +1145,101 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ *  Determines the appropriate autoload value for an option based on input.
+ *
+ *  This function checks the provided autoload value and returns a standardized value
+ *  ('on', 'off', 'auto-on', 'auto-off', or 'auto') based on specific conditions.
+ *
+ * If no explicit autoload value is provided, the function will check for certain heuristics around the given option.
+ * It will return `auto-on` to indicate autoloading, `auto-off` to indicate not autoloading, or `auto` if no clear
+ * decision could be made.
+ *
+ * @since 6.6.0
+ * @access private
+ *
+ * @param string $option          The name of the option.
+ * @param mixed $value            The value of the option to check its autoload value.
+ * @param mixed $serialized_value The serialized value of the option to check its autoload value.
+ * @param bool|null $autoload     The autoload value to check.
+ *                                Accepts 'on'|true to enable or 'off'|false to disable, or
+ *                                'auto-on', 'auto-off', or 'auto' for internal purposes.
+ *                                Any other autoload value will be forced to either 'auto-on',
+ *                                'auto-off', or 'auto'.
+ *                                'yes' and 'no' are supported for backward compatibility.
+ * @return string Returns the original $autoload value if explicit, or 'auto-on', 'auto-off',
+ *                or 'auto' depending on default heuristics.
+ */
+function wp_determine_option_autoload_value( $option, $value, $serialized_value, $autoload ) {
+
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'on' : 'off';
+	}
+
+	switch ( $autoload ) {
+		case 'on':
+		case 'yes':
+			return 'on';
+		case 'off':
+		case 'no':
+			return 'off';
+	}
+
+	// For backward compatibility, any formally unsupported value is interpreted as 'on' (formerly 'yes').
+	if ( null !== $autoload ) {
+		return 'on';
+	}
```

Comment:
I just realized we should probably also allow `auto`, `auto-on` and `auto-off` here to fall through, just to be safe. Those values are currently never provided to the function, but it's just safer to include that logic to handle it gracefully.

Any "auto" value means it may be overwritten by this function, so those values should go through the filter below. They must not become `'on'`.
```suggestion
	/*
	 * For backward compatibility, any formally unsupported value is interpreted as 'on' (formerly 'yes').
	 * The values assigned by WordPress core ("auto", "auto-on", "auto-off") may be overwritten as they
	 * are recalculated using the filter below.
	 */
	if ( ! in_array( $autoload, array( null, 'auto', 'auto-on', 'auto-off' ), true ) {
		return 'on';
	}
```